From 7dab274a8deb93f84e436b2f6ba561f83615dc79 Mon Sep 17 00:00:00 2001
From: GloriousEggroll <gloriouseggroll@gmail.com>
Date: Mon, 16 Mar 2026 22:09:03 -0600
Subject: [PATCH] decouple wineopenxr from steamvr and integrate it into wine
 directly

---
 configure.ac                           |    7 +
 loader/wine.inf.in                     |    1 +

diff --git a/configure.ac b/configure.ac
index 40078cf2a5b..ecfdb315d38 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2398,6 +2398,7 @@ dnl *** Check for modules to disable by default
 
 enable_win16=${enable_win16:-i386}
 enable_vcruntime140_1=${enable_vcruntime140_1:-x86_64,arm64ec,aarch64}
+enable_wineopenxr=${enable_wineopenxr:-x86_64,arm64ec,aarch64}
 
 if test -n "$PE_ARCHS"
 then
@@ -3342,6 +3343,12 @@ WINE_CONFIG_MAKEFILE(dlls/winegstreamer)
 WINE_CONFIG_MAKEFILE(dlls/winehid.sys)
 WINE_CONFIG_MAKEFILE(dlls/winemac.drv)
 WINE_CONFIG_MAKEFILE(dlls/winemapi)
+WINE_CONFIG_MAKEFILE(dlls/wineopenxr)
+if test "$HOST_ARCH" = "i386"
+then
+    DISABLED_SUBDIRS="$DISABLED_SUBDIRS dlls/wineopenxr"
+fi
+
 WINE_CONFIG_MAKEFILE(dlls/wineoss.drv)
 WINE_CONFIG_MAKEFILE(dlls/wineps.drv)
 WINE_CONFIG_MAKEFILE(dlls/wineps16.drv16,enable_win16)
diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 1b822ad1eab..9fe8a5f5656 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -2282,6 +2282,7 @@ HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.42.344
 11,,vidreszr.dll,1
 11,,windowscodecs.dll,1
 11,,winegstreamer.dll,1
+11,,wineopenxr.dll,1
 55,,wineps.drv,1
 11,,winevulkan.dll,1
 55,,winprint.dll,1
-- 
2.53.0
