From 02bb0a34ad51a7ac4eda1f50d6ac3b7938487185 Mon Sep 17 00:00:00 2001
From: Hans Leidekker <hans@codeweavers.com>
Date: Mon, 16 Mar 2026 13:28:43 +0100
Subject: [PATCH] crypt32: Trace CERT_CHAIN_ENGINE_CONFIG fields in
 CertCreateCertificateChainEngine().

---
 dlls/crypt32/chain.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/dlls/crypt32/chain.c b/dlls/crypt32/chain.c
index 1b38e789f159..6e0e55b9e5a4 100644
--- a/dlls/crypt32/chain.c
+++ b/dlls/crypt32/chain.c
@@ -227,6 +227,19 @@ BOOL WINAPI CertCreateCertificateChainEngine(PCERT_CHAIN_ENGINE_CONFIG pConfig,
     BOOL ret;
 
     TRACE("(%p, %p)\n", pConfig, phChainEngine);
+    TRACE("cbSize %lu\n", pConfig->cbSize);
+    TRACE("hRestrictedRoot %p\n", pConfig->hRestrictedRoot);
+    TRACE("hRestrictedTrust %p\n", pConfig->hRestrictedTrust);
+    TRACE("hRestrictedOther %p\n", pConfig->hRestrictedOther);
+    TRACE("cAdditionalStore %lu\n", pConfig->cAdditionalStore);
+    TRACE("dwFlags %lx\n", pConfig->dwFlags);
+    TRACE("dwUrlRetrievalTimeout %lu\n", pConfig->dwUrlRetrievalTimeout);
+    TRACE("MaximumCachedCertificates %lu\n", pConfig->MaximumCachedCertificates);
+    TRACE("CycleDetectionModulus %lu\n", pConfig->CycleDetectionModulus);
+    TRACE("hExclusiveRoot %p\n", pConfig->hExclusiveRoot);
+    TRACE("hExclusiveTrustedPeople %p\n", pConfig->hExclusiveTrustedPeople);
+    TRACE("dwExclusiveFlags %lx\n", pConfig->dwExclusiveFlags);
+    if (pConfig->dwExclusiveFlags) FIXME("dwExclusiveFlags %lx not supported\n", pConfig->dwExclusiveFlags);
 
     if (pConfig->cbSize != sizeof(CERT_CHAIN_ENGINE_CONFIG_NO_EXCLUSIVE_ROOT)
      && pConfig->cbSize != sizeof(CERT_CHAIN_ENGINE_CONFIG))
