diff --git a/dlls/kernelbase/kernelbase.spec b/dlls/kernelbase/kernelbase.spec
index 8f4a66a96fa..4123532897a 100644
--- a/dlls/kernelbase/kernelbase.spec
+++ b/dlls/kernelbase/kernelbase.spec
@@ -552,7 +552,7 @@
 @ stdcall GetFileTime(long ptr ptr ptr)
 @ stdcall GetFileType(long)
 @ stdcall GetFileVersionInfoA(str long long ptr)
-# @ stub GetFileVersionInfoByHandle
+@ stdcall GetFileVersionInfoByHandle(long long long ptr)
 @ stdcall GetFileVersionInfoExA(long str long long ptr)
 @ stdcall GetFileVersionInfoExW(long wstr long long ptr)
 @ stdcall GetFileVersionInfoSizeA(str ptr)
diff --git a/dlls/kernelbase/version.c b/dlls/kernelbase/version.c
index 2690ac65f9a..15482e81f5d 100644
--- a/dlls/kernelbase/version.c
+++ b/dlls/kernelbase/version.c
@@ -1006,6 +1006,17 @@ BOOL WINAPI GetFileVersionInfoA( LPCSTR filename, DWORD handle, DWORD datasize,
     return GetFileVersionInfoExA(FILE_VER_GET_LOCALISED, filename, handle, datasize, data);
 }
 
+/***********************************************************************
+ *           GetFileVersionInfoByHandle        (kernelbase.@)
+ */
+BOOL WINAPI GetFileVersionInfoByHandle( DWORD flags, DWORD handle, DWORD datasize, LPVOID data )
+{
+    FIXME("(%#lx,%#lx,%lu,%p): stub\n", flags, handle, datasize, data);
+
+    SetLastError( ERROR_CALL_NOT_IMPLEMENTED );
+    return FALSE;
+}
+
 /***********************************************************************
  *           VersionInfo16_FindChild             [internal]
  */
diff --git a/dlls/version/version.spec b/dlls/version/version.spec
index a3a9955e02c..243dcc4b606 100644
--- a/dlls/version/version.spec
+++ b/dlls/version/version.spec
@@ -1,4 +1,5 @@
 @ stdcall -import GetFileVersionInfoA(str long long ptr)
+@ stdcall -import GetFileVersionInfoByHandle(long long long ptr)
 @ stdcall -import GetFileVersionInfoExA(long str long long ptr)
 @ stdcall -import GetFileVersionInfoExW(long wstr long long ptr)
 @ stdcall -import GetFileVersionInfoSizeA(str ptr)
diff --git a/include/winver.h b/include/winver.h
index 007877e8e4d..f3f5b9ca622 100644
--- a/include/winver.h
+++ b/include/winver.h
@@ -50,6 +50,7 @@ WINBASEAPI DWORD WINAPI GetFileVersionInfoSizeW(LPCWSTR,LPDWORD);
 WINBASEAPI BOOL  WINAPI GetFileVersionInfoA(LPCSTR,DWORD,DWORD,LPVOID);
 WINBASEAPI BOOL  WINAPI GetFileVersionInfoW(LPCWSTR,DWORD,DWORD,LPVOID);
 #define                 GetFileVersionInfo WINELIB_NAME_AW(GetFileVersionInfo)
+WINBASEAPI BOOL  WINAPI GetFileVersionInfoByHandle(DWORD,DWORD,DWORD,LPVOID);
 WINBASEAPI DWORD WINAPI GetFileVersionInfoSizeExA(DWORD,LPCSTR,LPDWORD);
 WINBASEAPI DWORD WINAPI GetFileVersionInfoSizeExW(DWORD,LPCWSTR,LPDWORD);
 #define                 GetFileVersionInfoSizeEx WINELIB_NAME_AW(GetFileVersionInfoSizeEx)
