diff --git a/dlls/d3drm/viewport.c b/dlls/d3drm/viewport.c
index 64f0dd97b61..66703bb5827 100644
--- a/dlls/d3drm/viewport.c
+++ b/dlls/d3drm/viewport.c
@@ -998,8 +998,6 @@ static HRESULT WINAPI d3drm_picked_array_GetPick(IDirect3DRMPickedArray *iface,
         DWORD index, IDirect3DRMVisual **visual, IDirect3DRMFrameArray **frame_array,
         D3DRMPICKDESC *pick_desc)
 {
-    struct d3drm_picked_array *array = impl_from_IDirect3DRMPickedArray(iface);
-
     TRACE("iface %p, index %lu, visual %p, frame_array %p, pick_desc %p.\n", iface, index,
           visual, frame_array, pick_desc);
 
