From 35a8766923e2047b3f17fe3a5fbf3c6933386173 Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Fri, 23 Jan 2026 22:31:11 +0100
Subject: [PATCH 003/113] winewayland: Fix build error with older wl_pointer
 interface.

---
 dlls/winewayland.drv/wayland_pointer.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winewayland.drv/wayland_pointer.c b/dlls/winewayland.drv/wayland_pointer.c
index f7e1b5c1a82..3fa30f9dffa 100644
--- a/dlls/winewayland.drv/wayland_pointer.c
+++ b/dlls/winewayland.drv/wayland_pointer.c
@@ -336,7 +336,9 @@ static const struct wl_pointer_listener pointer_listener =
     pointer_handle_axis_source,
     pointer_handle_axis_stop,
     pointer_handle_axis_discrete,
+#ifdef WL_POINTER_AXIS_VALUE120_SINCE_VERSION
     pointer_handle_axis_value120
+#endif
 };
 
 /**********************************************************************
-- 
2.54.0

