From d877f2303f468436a68c7e93c1d01d76c38ad7a2 Mon Sep 17 00:00:00 2001
From: Ziia Shi <mkrsym1@gmail.com>
Date: Tue, 3 Feb 2026 20:54:24 +0100
Subject: [PATCH 008/153] win32u: Call SetIMEEnabled from
 NtUserAssociateInputContext.

---
 dlls/win32u/imm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/win32u/imm.c b/dlls/win32u/imm.c
index 3337f66e291..07b41959442 100644
--- a/dlls/win32u/imm.c
+++ b/dlls/win32u/imm.c
@@ -221,6 +221,9 @@ UINT WINAPI NtUserAssociateInputContext( HWND hwnd, HIMC ctx, ULONG flags )
     }
 
     release_win_ptr( win );
+
+    user_driver->pSetIMEEnabled( hwnd, ctx != NULL );
+
     return ret;
 }
 
-- 
2.54.0

