From c6eefae16c7e23224636950898326fbca03e51e5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 23 Jun 2026 11:07:33 +0200
Subject: [PATCH 010/239] winewayland: Make sure the primary monitor rect is at
 0x0.

---
 dlls/winewayland.drv/display.c | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/dlls/winewayland.drv/display.c b/dlls/winewayland.drv/display.c
index b2a94289abb..9ed85671745 100644
--- a/dlls/winewayland.drv/display.c
+++ b/dlls/winewayland.drv/display.c
@@ -206,13 +206,15 @@ static void wayland_add_device_source(const struct gdi_device_manager *device_ma
 }
 
 static void wayland_add_device_monitor(const struct gdi_device_manager *device_manager,
-                                       void *param, struct output_info *output_info)
+                                       void *param, struct output_info *output_info,
+                                       struct output_info *primary)
 {
     struct gdi_monitor monitor = {0};
 
     SetRect(&monitor.rc_monitor, output_info->x, output_info->y,
             output_info->x + output_info->output->current_mode->width,
             output_info->y + output_info->output->current_mode->height);
+    OffsetRect(&monitor.rc_monitor, -primary->x, -primary->y);
 
     /* We don't have a direct way to get the work area in Wayland. */
     monitor.rc_work = monitor.rc_monitor;
@@ -237,7 +239,8 @@ static void populate_devmode(struct wayland_output_mode *output_mode, DEVMODEW *
 }
 
 static void wayland_add_device_modes(const struct gdi_device_manager *device_manager,
-                                     void *param, struct output_info *output_info)
+                                     void *param, struct output_info *output_info,
+                                     struct output_info *primary)
 {
     DEVMODEW *modes, current = {.dmSize = sizeof(current)};
     struct wayland_output_mode *output_mode;
@@ -248,8 +251,8 @@ static void wayland_add_device_modes(const struct gdi_device_manager *device_man
 
     populate_devmode(output_info->output->current_mode, &current);
     current.dmFields |= DM_POSITION;
-    current.dmPosition.x = output_info->x;
-    current.dmPosition.y = output_info->y;
+    current.dmPosition.x = output_info->x - primary->x;
+    current.dmPosition.y = output_info->y - primary->y;
 
     RB_FOR_EACH_ENTRY(output_mode, &output_info->output->modes,
                       struct wayland_output_mode, entry)
@@ -270,8 +273,8 @@ UINT WAYLAND_UpdateDisplayDevices(const struct gdi_device_manager *device_manage
 {
     struct wayland_output *output;
     DWORD state_flags = DISPLAY_DEVICE_ATTACHED_TO_DESKTOP | DISPLAY_DEVICE_PRIMARY_DEVICE;
+    struct output_info *primary = NULL, *output_info;
     struct wl_array output_info_array;
-    struct output_info *output_info;
 
     TRACE("\n");
 
@@ -294,9 +297,10 @@ UINT WAYLAND_UpdateDisplayDevices(const struct gdi_device_manager *device_manage
 
     wl_array_for_each(output_info, &output_info_array)
     {
+        if (!primary) primary = output_info;
         wayland_add_device_source(device_manager, param, state_flags, output_info);
-        wayland_add_device_monitor(device_manager, param, output_info);
-        wayland_add_device_modes(device_manager, param, output_info);
+        wayland_add_device_monitor(device_manager, param, output_info, primary);
+        wayland_add_device_modes(device_manager, param, output_info, primary);
         state_flags &= ~DISPLAY_DEVICE_PRIMARY_DEVICE;
     }
 
-- 
2.55.0

