From fcb1d7f62899ef7ef0c4dfb22a98b550d3c2de20 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Mon, 15 Jun 2026 21:39:23 -0400
Subject: [PATCH 016/268] winewayland: Fix missing wayland_win_data_release.

---
 dlls/winewayland.drv/wayland_surface.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index 4a7dfbb042c..1d100a1f304 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -170,6 +170,8 @@ void wp_fractional_scale_handle_scale(void* user_data,
         surface->processing.processed = TRUE;
     }
 
+    wayland_win_data_release(data);
+
     NtUserExposeWindowSurface(hwnd, 0, NULL, 0);
 }
 
-- 
2.55.0

