From aa368df913c783480ce58be0d37d77b407f99bf1 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Fri, 29 May 2026 17:32:24 -0400
Subject: [PATCH 134/153] ime: Reduce log spam of ImeSetCompositionString.

---
 dlls/imm32/ime.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/imm32/ime.c b/dlls/imm32/ime.c
index 766e144eb3e..2f3e633f03a 100644
--- a/dlls/imm32/ime.c
+++ b/dlls/imm32/ime.c
@@ -635,10 +635,11 @@ DWORD WINAPI ImeConversionList( HIMC himc, const WCHAR *source, CANDIDATELIST *d
 BOOL WINAPI ImeSetCompositionString( HIMC himc, DWORD index, const void *comp, DWORD comp_len,
                                      const void *read, DWORD read_len )
 {
+    static int once;
     INPUTCONTEXT *ctx;
 
-    FIXME( "himc %p, index %lu, comp %p, comp_len %lu, read %p, read_len %lu semi-stub!\n",
-            himc, index, comp, comp_len, read, read_len );
+    if (!once++) FIXME( "himc %p, index %lu, comp %p, comp_len %lu, read %p, read_len %lu semi-stub!\n",
+                         himc, index, comp, comp_len, read, read_len );
     if (read && read_len) FIXME( "Read string unimplemented\n" );
     if (index != SCS_SETSTR && index != SCS_CHANGECLAUSE && index != SCS_CHANGEATTR) return FALSE;
 
-- 
2.54.0

