From 033a257acca74efd5528607ad1b7ed7be8ea2655 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Fri, 29 May 2026 17:31:25 -0400
Subject: [PATCH 136/148] win32u: Reduce log spam of
 NtUserActivateKeyboardLayout.

---
 dlls/win32u/input.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index aa2d5ebb260..c6d356061cf 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -1383,10 +1383,11 @@ HKL WINAPI NtUserActivateKeyboardLayout( HKL layout, UINT flags )
     HKL old_layout;
     LCID locale;
     HWND focus;
+    static int once;
 
     TRACE_(keyboard)( "layout %p, flags %x\n", layout, flags );
 
-    if (flags) FIXME_(keyboard)( "flags %x not supported\n", flags );
+    if (flags && !once++) FIXME_(keyboard)( "flags %x not supported\n", flags );
 
     if (layout == (HKL)HKL_NEXT || layout == (HKL)HKL_PREV)
     {
-- 
2.54.0

