From e37658af794a6138ae8b27ea1401ccad9502d00d Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Wed, 24 Jun 2026 03:48:18 -0400
Subject: [PATCH 180/239] server: Use MOUSEEVENTF_MOVE if the cursor win
 changes.

---
 server/queue.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/server/queue.c b/server/queue.c
index a5b8c73117b..32f76c83154 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -2307,7 +2307,8 @@ static int queue_mouse_message( struct desktop *desktop, user_handle_t win, cons
             x = desktop_shm->cursor.x + input->mouse.x;
             y = desktop_shm->cursor.y + input->mouse.y;
         }
-        if (x == desktop_shm->cursor.x && y == desktop_shm->cursor.y)
+        if (x == desktop_shm->cursor.x && y == desktop_shm->cursor.y &&
+            (!win || desktop->cursor_win == win))
             flags &= ~MOUSEEVENTF_MOVE;
     }
     else
-- 
2.55.0

