From a33099db5b81ed864d8e4323a05387b5dc032898 Mon Sep 17 00:00:00 2001
From: bk304 <xnep1@hotmail.com>
Date: Thu, 23 Jul 2026 12:15:56 -0300
Subject: [PATCH 227/239] winewayland: Removed redundant scan2vk_brazil layout

---
 dlls/winewayland.drv/wayland_keyboard.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_keyboard.c b/dlls/winewayland.drv/wayland_keyboard.c
index 8bf449aa3d6..2edee2ad6eb 100644
--- a/dlls/winewayland.drv/wayland_keyboard.c
+++ b/dlls/winewayland.drv/wayland_keyboard.c
@@ -151,15 +151,6 @@ static const USHORT scan2vk_dvorak[0x280] =
     EXTRA_SCAN2VK
 };
 
-static const USHORT scan2vk_brazil[0x280] =
-{
-    T00, T01, T02, T03, T04, T05, T06, T07, T08, T09, T0A, T0B, T0C, T0D, T0E,
-    T0F, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T1A, T1B, T1C,
-    T1D, T1E, T1F, T20, T21, T22, T23, T24, T25, T26, T27, T28, T29,
-    T2A, VK_OEM_102, T2C, T2D, T2E, T2F, T30, T31, T32, T33, T34, T35,
-    EXTRA_SCAN2VK
-};
-
 static WORD key2scan(UINT key)
 {
     /* base keys can be mapped directly */
@@ -509,7 +500,6 @@ static void add_xkb_layout(const char *xkb_layout, struct xkb_keymap *xkb_keymap
     case MAKELANGID(LANG_FRENCH, SUBLANG_DEFAULT): scan2vk = scan2vk_azerty; break;
     case MAKELANGID(LANG_GERMAN, SUBLANG_DEFAULT): scan2vk = scan2vk_qwertz; break;
     case MAKELANGID(LANG_GERMAN, SUBLANG_GERMAN_SWISS): scan2vk = scan2vk_qwertz; break;
-    case MAKELANGID(LANG_PORTUGUESE, SUBLANG_PORTUGUESE_BRAZILIAN): scan2vk = scan2vk_brazil; break;
     default: scan2vk = scan2vk_qwerty; break;
     }
     if (strstr(xkb_layout, "dvorak")) scan2vk = scan2vk_dvorak;
-- 
2.55.0

