From e064631b09fccc74d2603f55fe419d10b525581e Mon Sep 17 00:00:00 2001
From: bk304 <xnep1@hotmail.com>
Date: Thu, 23 Jul 2026 12:17:10 -0300
Subject: [PATCH 228/239] winewayland: Fix misconfigured portuguese sublanguage

---
 dlls/winewayland.drv/wayland_keyboard.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_keyboard.c b/dlls/winewayland.drv/wayland_keyboard.c
index 2edee2ad6eb..88f8a62177d 100644
--- a/dlls/winewayland.drv/wayland_keyboard.c
+++ b/dlls/winewayland.drv/wayland_keyboard.c
@@ -310,7 +310,7 @@ static inline LANGID langid_from_xkb_layout(const char *layout, size_t layout_le
         [MAKEINDEX('b','d')] = MAKELANGID(LANG_BANGLA, SUBLANG_DEFAULT),
         [MAKEINDEX('b','e')] = MAKELANGID(LANG_FRENCH, SUBLANG_FRENCH_BELGIAN),
         [MAKEINDEX('b','g')] = MAKELANGID(LANG_BULGARIAN, SUBLANG_DEFAULT),
-        [MAKEINDEX('b','r')] = MAKELANGID(LANG_PORTUGUESE, 2),
+        [MAKEINDEX('b','r')] = MAKELANGID(LANG_PORTUGUESE, SUBLANG_PORTUGUESE_BRAZILIAN),
         [MAKEINDEX('b','t')] = MAKELANGID(LANG_TIBETAN, 3),
         [MAKEINDEX('b','w')] = MAKELANGID(LANG_TSWANA, SUBLANG_TSWANA_BOTSWANA),
         [MAKEINDEX('b','y')] = MAKELANGID(LANG_BELARUSIAN, SUBLANG_DEFAULT),
@@ -371,7 +371,7 @@ static inline LANGID langid_from_xkb_layout(const char *layout, size_t layout_le
         [MAKEINDEX('p','h')] = MAKELANGID(LANG_FILIPINO, SUBLANG_DEFAULT),
         [MAKEINDEX('p','k')] = MAKELANGID(LANG_URDU, SUBLANG_DEFAULT),
         [MAKEINDEX('p','l')] = MAKELANGID(LANG_POLISH, SUBLANG_DEFAULT),
-        [MAKEINDEX('p','t')] = MAKELANGID(LANG_PORTUGUESE, SUBLANG_DEFAULT),
+        [MAKEINDEX('p','t')] = MAKELANGID(LANG_PORTUGUESE, SUBLANG_PORTUGUESE),
         [MAKEINDEX('r','o')] = MAKELANGID(LANG_ROMANIAN, SUBLANG_DEFAULT),
         [MAKEINDEX('r','s')] = MAKELANGID(LANG_SERBIAN, SUBLANG_SERBIAN_LATIN),
         [MAKEINDEX('r','u')] = MAKELANGID(LANG_RUSSIAN, SUBLANG_DEFAULT),
-- 
2.55.0

