From dbaf6853add884bd133950e1f8edd01a5b96c0f5 Mon Sep 17 00:00:00 2001
From: bk304 <xnep1@hotmail.com>
Date: Thu, 23 Jul 2026 12:18:22 -0300
Subject: [PATCH 229/239] winewayland: Add KEY_KPCOMMA to scancode translation

---
 dlls/winewayland.drv/wayland_keyboard.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winewayland.drv/wayland_keyboard.c b/dlls/winewayland.drv/wayland_keyboard.c
index 88f8a62177d..f43a3604784 100644
--- a/dlls/winewayland.drv/wayland_keyboard.c
+++ b/dlls/winewayland.drv/wayland_keyboard.c
@@ -189,7 +189,7 @@ static WORD key2scan(UINT key)
     case KEY_F24: return 0x0076; /* T76 / VK_F24 */
     case KEY_KPPLUSMINUS: return 0x007b; /* T7B / VK_OEM_PA1 */
     /* FIXME: map a KEY to T7C / VK_TAB */
-    /* FIXME: map a KEY to T7E / VK_ABNT_C2 */
+    case KEY_KPCOMMA: return 0x007E; /* T7E / VK_ABNT_C2 */
     /* FIXME: map a KEY to T7F / VK_OEM_PA2 */
     case KEY_PREVIOUSSONG: return 0x0110; /* X10 / VK_MEDIA_PREV_TRACK */
     case KEY_NEXTSONG: return 0x0119; /* X19 / VK_MEDIA_NEXT_TRACK */
-- 
2.55.0

