From d98f4d69b8f62708f2a252143044d308a9f5ce7a Mon Sep 17 00:00:00 2001
From: bk304 <xnep1@hotmail.com>
Date: Thu, 23 Jul 2026 13:46:17 -0300
Subject: [PATCH 230/239] winewayland: Skip invalid keycode mappings.

---
 dlls/winewayland.drv/wayland_keyboard.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winewayland.drv/wayland_keyboard.c b/dlls/winewayland.drv/wayland_keyboard.c
index f43a3604784..251919c0374 100644
--- a/dlls/winewayland.drv/wayland_keyboard.c
+++ b/dlls/winewayland.drv/wayland_keyboard.c
@@ -599,6 +599,8 @@ static void add_xkb_layout(const char *xkb_layout, struct xkb_keymap *xkb_keymap
         uint32_t caps_ret, shift_ret;
         unsigned int mod;
 
+        if (!vkey) continue;
+
         if ((vkey & KBDNUMPAD) && (vkey & 0xff) == VK_DELETE)
         {
             VK_TO_WCHARS8 num_vkey2wch = {.VirtualKey = VK_DECIMAL};
-- 
2.55.0

