From 6f925a3d23e399705a48efe2ca20263380ebb4fe Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Sat, 25 Jul 2026 02:39:08 -0400
Subject: [PATCH 235/268] d3dkmt: Implement KMTQAITYPE_UMDRIVERNAME

---
 dlls/win32u/d3dkmt.c | 44 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/dlls/win32u/d3dkmt.c b/dlls/win32u/d3dkmt.c
index 3e69cd3dc68..d8b69ceea21 100644
--- a/dlls/win32u/d3dkmt.c
+++ b/dlls/win32u/d3dkmt.c
@@ -705,6 +705,21 @@ typedef struct
     } info[1];
 } AMDLargeDriverPrivate;
 
+typedef enum _KMTUMDVERSION {
+  KMTUMDVERSION_DX9,
+  KMTUMDVERSION_DX10,
+  KMTUMDVERSION_DX11,
+  KMTUMDVERSION_DX12,
+  KMTUMDVERSION_DX12_WSA32,
+  KMTUMDVERSION_DX12_WSA64,
+  NUM_KMTUMDVERSIONS
+} KMTUMDVERSION;
+
+typedef struct _D3DKMT_UMDFILENAMEINFO {
+    KMTUMDVERSION Version;
+    WCHAR         UmdFileName[MAX_PATH];
+} D3DKMT_UMDFILENAMEINFO;
+
 /******************************************************************************
  *           NtGdiDdDDIQueryAdapterInfo    (win32u.@)
  */
@@ -862,6 +877,35 @@ NTSTATUS WINAPI NtGdiDdDDIQueryAdapterInfo( D3DKMT_QUERYADAPTERINFO *desc )
         FIXME("Unsupported KMTQAITYPE_UMDRIVERPRIVATE!\n");
         return STATUS_NOT_IMPLEMENTED;
     }
+    case KMTQAITYPE_UMDRIVERNAME:
+    {
+        VkPhysicalDeviceProperties2KHR properties2 = {0};
+        struct vulkan_physical_device *physical_device;
+        struct vulkan_instance *instance;
+        D3DKMT_UMDFILENAMEINFO *info = desc->pPrivateDriverData;
+
+        if (desc->PrivateDriverDataSize < sizeof(*info)) return STATUS_INVALID_PARAMETER;
+        if (info->Version >= NUM_KMTUMDVERSIONS) return STATUS_INVALID_PARAMETER;
+
+        TRACE("Version %u\n", info->Version);
+
+        if (!(adapter = get_d3dkmt_object( desc->hAdapter, D3DKMT_ADAPTER ))) return STATUS_INVALID_PARAMETER;
+        if (!(physical_device = adapter->physical_device)) return STATUS_INVALID_PARAMETER;
+        instance = physical_device->instance;
+
+        properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
+        instance->p_vkGetPhysicalDeviceProperties2KHR( physical_device->host.physical_device, &properties2 );
+
+        if (properties2.properties.vendorID == 0x1002 && info->Version >= KMTUMDVERSION_DX12)
+        {
+            asciiz_to_unicode(info->UmdFileName, "C:\\Windows\\System32\\amdxc64.dll");
+            return STATUS_SUCCESS;
+        }
+
+        FIXME("KMTQAITYPE_UMDRIVERNAME\n");
+
+        return STATUS_NOT_IMPLEMENTED;
+    }
     default:
     {
         FIXME( "type %d not handled.\n", desc->Type );
-- 
2.55.0

