diff --git a/dlls/comctl32/animate.c b/dlls/comctl32/animate.c
index 2085115187d..acab1141610 100644
--- a/dlls/comctl32/animate.c
+++ b/dlls/comctl32/animate.c
@@ -619,8 +619,6 @@ static BOOL ANIMATE_GetAviInfo(ANIMATE_INFO *infoPtr)
 
 static BOOL ANIMATE_GetAviCodec(ANIMATE_INFO *infoPtr)
 {
-    DWORD	outSize;
-
     /* check uncompressed AVI */
     if ((infoPtr->ash.fccHandler == mmioFOURCC('D', 'I', 'B', ' ')) ||
        (infoPtr->ash.fccHandler == mmioFOURCC('R', 'L', 'E', ' ')) ||
