From ec098281054467a5d680a9819a96a028bdc94673 Mon Sep 17 00:00:00 2001
From: GloriousEggroll <gloriouseggroll@gmail.com>
Date: Sat, 13 Jun 2026 04:04:56 -0600
Subject: [PATCH] fix video playback in tokyo necro

---
 dlls/quartz/filtergraph.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/quartz/filtergraph.c b/dlls/quartz/filtergraph.c
index 1da175f0332..2de13f5d6d9 100644
--- a/dlls/quartz/filtergraph.c
+++ b/dlls/quartz/filtergraph.c
@@ -2078,7 +2078,7 @@ static void CALLBACK wait_pause_cb(TP_CALLBACK_INSTANCE *instance, void *context
     OAFilterState state;
     HRESULT hr;
 
-    if ((hr = IMediaControl_GetState(control, INFINITE, &state)) != S_OK)
+    if ((hr = IMediaControl_GetState(control, 100, &state)) != S_OK && hr != VFW_S_STATE_INTERMEDIATE)
         ERR("Failed to get paused state, hr %#lx.\n", hr);
 
     if (FAILED(hr = IMediaControl_Stop(control)))
-- 
2.54.0

